### ISSUE TITLE
[TTS] Not all available languages are included on the app

### ISSUE THREAD

**Issue by matheusgomesms** [organicmaps/organicmaps#5032]
_**Describe the issue**
I used heavily OM this past month and for some reason it seems TTS instructions in PT-BR were using PT strings (or maybe old PT-BR strings which I know I corrected long time ago). I checked data/strings/sound.txt and data/sound-strings/pt-BR.json/localize.json; both are correct._

_**System information:**
iOS last version._

Update:

**I think I discovered the actual problem. [Some languages available on the code](https://github.com/organicmaps/organicmaps/tree/master/data/sound-strings) are actually not available on the app! be, es-mx, eu, nb, pt-br, sw and probably some others (didn't check all).**

**Comment by matheusgomesms** [comment]
I also improved TTS in PT-BR, per #5033.

**Comment by RicoElectrico** [comment]
On Samsung Galaxy S8 also I can't see all languages.
Not sure if I should file another bug. The issue is that Samsung enables its own TTS by default which only supports a handful of languages. Even though Google TTS is present and has other languages, OMaps can't find it when it's not selected as Android default.

**Comment by biodranik** [comment]
At the moment, Organic Maps can use only one, default engine. It would be great to fix the code and support all engines that are available.

**Comment by matheusgomesms** [comment]
Good to know it's not only on iOS.

@biodranik Regarding languages that are probably available in all major TTS engines (like PT-BR), how it can be fixed? I can gladly provide a PR.

I saw something related to android/res/values/strings-tts.xml, platform/languages.hpp and maybe indexer/categories_holder.hpp. Is there anything else I should know?

**Comment by matheusgomesms** [comment]
For Chinese (which has variants) I saw some exceptions in the code, not sure if it's the same case as PT/PT-BR (or ES/ES-MX), for example.

**Comment by matheusgomesms** [comment]
I would like to reopen this issue. On iOS, the only language added was Norwegian from that list. Not sure on Android. Still missing these languages (which theoretically are available on TTS):

Basque
Marathi
Persian
Portuguese (Brazil)
Spanish (Mexico)

It is also lacking Belorussian and Swahili, but I’m not sure they are available on iOS for TTS.

**Comment by biodranik** [comment]
@matheusgomesms are you checking the latest master branch/AppStore version? Is there something suspicious in logs?

**Comment by matheusgomesms** [comment]
Yes, I’m using the latest AppStore version (from a couple of days ago). The app log to file a bug doesn’t say much:

- Organic Maps 2023.05.08-1
- pt-BR
- pt-BR, en-BR, es-BR 

(these last line shows my preferred languages on iOS settings).

Tested both on an iPhone and an iPad.

**Comment by matheusgomesms** [comment]
These are the available TTS languages. For app languages, everything is fine.

![IMG_7270](https://github.com/organicmaps/organicmaps/assets/86851490/e5a2b16f-f110-4def-8d49-431364e42cd0)
![IMG_7271](https://github.com/organicmaps/organicmaps/assets/86851490/e1ea5c25-23dc-4803-8571-20ce68e9d8e8)

**Comment by matheusgomesms** [comment]
Ok, silly of me. The aforementioned languages were not installed by default on iOS for some reason (while all the rest are).

So, the only languages missing are ES-MX and PT-BR. Even though I select PT-BR on iOS settings, it is not used.

Perhaps that special setting for Chinese (simplified and traditional) has to be used also for these languages with variants?
